scala-package/core/src/main/scala/ml/dmlc/mxnet/IO.scala

```diff
@@ -95,8 +95,8 @@ object IO {
   private[mxnet] def initData(data: IndexedSeq[NDArray],
                               allowEmpty: Boolean,
                               defaultName: String): IndexedSeq[(String, NDArray)] = {
-    require(data != null || allowEmpty)
-    if (data == null) {
```

Comment:
I think we shall keep checking `null`, i.e., `require((data != null && !data.isEmpty) || allowEmpty)`, in case users do pass it in. Same for the other modifications.